diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5bad53988..e4dc07c1f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,6 +29,10 @@ set (CMAKE_PROJECT_VERSION_MAJOR ${GTSAM_VERSION_MAJOR})
 set (CMAKE_PROJECT_VERSION_MINOR ${GTSAM_VERSION_MINOR})
 set (CMAKE_PROJECT_VERSION_PATCH ${GTSAM_VERSION_PATCH})
 
+if(MSVC)
+  add_compile_options(/Zm200)
+endif()
+
 ###############################################################################
 # Gather information, perform checks, set defaults
 
