diff --git a/gtsam/navigation/tests/testCombinedImuFactor.cpp b/gtsam/navigation/tests/testCombinedImuFactor.cpp
index 7fd1d8db7..d99ce8306 100644
--- a/gtsam/navigation/tests/testCombinedImuFactor.cpp
+++ b/gtsam/navigation/tests/testCombinedImuFactor.cpp
@@ -197,8 +197,8 @@ TEST_PIM(CombinedImuFactor, PredictRotation) {
   const CombinedImuFactorT<CombinedPIM> Combinedfactor(X(1), V(1), X(2), V(2), B(1), B(2), pim);
 
   // Predict
-  const Pose3 x(Rot3::Ypr(0, 0, 0), Point3(0, 0, 0)), x2;
-  const Vector3 v(0, 0, 0), v2;
+  const Pose3 x(Rot3::Ypr(0, 0, 0), Point3(0, 0, 0));
+  const Vector3 v(0, 0, 0);
   const NavState actual = pim.predict(NavState(x, v), bias);
   const Pose3 expectedPose(Rot3::Ypr(M_PI / 10, 0, 0), Point3(0, 0, 0));
   EXPECT(assert_equal(expectedPose, actual.pose(), tol));
