diff --git a/gtsam_unstable/CMakeLists.txt b/gtsam_unstable/CMakeLists.txt
index 98a1b4ef9..f556563a9 100644
--- a/gtsam_unstable/CMakeLists.txt
+++ b/gtsam_unstable/CMakeLists.txt
@@ -85,6 +85,17 @@ set_target_properties(gtsam_unstable PROPERTIES
 target_link_libraries(gtsam_unstable PUBLIC gtsam)
 # No need to link against Boost here, it's inherited from gtsam PUBLIC interface
 
+# Append Eigen include path, set in top-level CMakeLists.txt to either
+# system-eigen, or GTSAM eigen path
+target_include_directories(gtsam_unstable PUBLIC
+  $<BUILD_INTERFACE:${GTSAM_EIGEN_INCLUDE_FOR_BUILD}>
+  $<INSTALL_INTERFACE:${GTSAM_EIGEN_INCLUDE_FOR_INSTALL}>
+)
+# MKL include dir:
+if (GTSAM_USE_EIGEN_MKL)
+  target_include_directories(gtsam_unstable PUBLIC ${MKL_INCLUDE_DIR})
+endif()
+
 if(WIN32) # Add 'lib' prefix to static library to avoid filename collision with shared library
        if (NOT BUILD_SHARED_LIBS)
                set_target_properties(gtsam_unstable PROPERTIES
